Assets/Hoppy/Scripts/PlayerController.cs

```diff
@@ -236,23 +245,53 @@ public void OnTriggerEnter(Collider col)
             numberOfPickUps++;
             PlayerPrefs.SetInt("NumberOfPickUps", numberOfPickUps);
 
-            // Update Number of Gems displayed.
-            uiController.updateNumberOfGemsUITexts();
-
             // Instantiate Gem Explosion in the same position of the picked Gem.
             GameObject gemExplosionObject = Instantiate(gemsExplosion, new Vector3(transform.position.x, gemsExplosion.transform.position.y, transform.position.z), gemsExplosion.transform.rotation) as GameObject;
 
             // Destroy Gem Explosion Object.
             Destroy(gemExplosionObject, 1f);
         }
-        if (col.tag == "Shield_Pickup")
+        else if (col.tag == "Shield_Item")
         {
+            // Show floating text
+            uiController.ShowFloatingText(col.gameObject,"Shield", itemColors[0]);
+
             // Collision detected with a Shield Object.
             // Deactivate The Shield Object.
             col.gameObject.SetActive(false);
+            
             // Call powerup handler
             enableShield();
         }
+        else if(col.tag == "Crate_Item")
```

Comment:
Eventually we are going to need a better handler for this instead of a bunch of else ifs. Not sure at the moment though but as we add more items we should think about a more elegant solution perhaps.